tests/netebpfext_unit/netebpf_ext_helper.cpp

```diff
@@ -116,6 +122,10 @@ _netebpf_ext_helper::~_netebpf_ext_helper()
         net_ebpf_ext_uninitialize_ndis_handles();
     }
 
+    if (random_initialized) {
```

Comment:
The constructor takes in the flag to indicate whether or not the platform needs to be initialized. If the flag is true, then this component will do the initialization. It will also be responsible for doing the cleanup. The newly added flag/check here ensures that we only do invoke _terminate() if this component did the initialization.

If we don't have the flag and we always invoke terminate(), we are incorrectly doing the cleanup for some other component which really initialized these helpers (and may be using them after we invoked terminate).